Source/DafnyCore/Resolver/NameResolutionAndTypeInference.cs

```diff
@@ -3268,8 +3271,10 @@ public XConstraint_EquatableArg(IToken tok, Type a, Type b, bool allowSuperSub,
           var ec = reporter.Count(ErrorLevel.Error);
           allTypeParameters.PushMarker();
           ResolveTypeParameters(function.TypeArgs, false, function);
+          DominatingStatementLabels.PushMarker();
```

Comment:
I moved these push and pop inside function.Resolve.
Yes both are needed because "requires labels" need to be accessible within the by method body if there is one.